### DISCUSSION TITLE
limit_execution configuration not working

### DISCUSSION THREAD

**Discussion by RAJMODI038** [jertel/elastalert2#1284] [Q&A] [answered]
yaml Configuration `limit_execution: "0 8 * * *"` to schedule the rule for everyday 8 AM is not working.
I've tested the same with `*/3 * * * *` without any error. Why the rule does not trigger? 
Can someone point out what could be wrong here?

```
name: test_alert_1
type: any
index: metrics_production-*
limit_execution: "0 8 * * *"
buffer_time:
  hours: 12
alert:
- debug
```

@jertel - After updating the log level, I got to know that elastalert rule did execute but it's not adding any entry in `_status` index!

**Comment by jertel** [comment]
I've not tried a rule without a `filter`. Perhaps try adding a simple filter to see if you can isolate the problem.

**Comment by RAJMODI038** [comment]
@jertel - I tried to debug the rule run and found that every time it's skipped due to this condition https://github.com/jertel/elastalert2/blob/6bcdd1ef4c5425cba5c4b7728de9cfd553e180ab/elastalert/elastalert.py#L1234-L1236. 
Here, I'm not overriding `run_every` configuration of config.yml (set to every 60 sec) as we already have limit_execution in the rule config.
Also, couldn't understand the comment:
```
            # If the estimated next endtime (end + run_every) isn't at least a minute past the next exec time
            # That means that we need to pause execution after this run
```
I've also noticed that `has_run_once` is always false since it's created and seems a dead-end. It's super complex for me to understand all the usecases taken care of here.

**Reply by jertel** [reply, marked as answer]
See https://github.com/jertel/elastalert2/discussions/590#discussioncomment-1829968

**Reply by RAJMODI038** [reply]
@jertel - Thanks a lot mahn! You're a lifesaver
